satpy/readers/yaml_reader.py

```diff
@@ -904,7 +1011,7 @@ def create_filehandlers(self, filenames, fh_kwargs=None):
         return created_fhs
 
     @staticmethod
-    def _load_dataset(dsid, ds_info, file_handlers, dim='y', pad_data=True):
+    def _load_dataset(dsid, ds_info, file_handlers, dim='y', pad_data=True, **kwargs):
```

Comment:
Is it possible to use explicit keyword arguments instead of `**kwargs`?

Reply:
I found this to be the least "invasive" solution to the  `got an unexpected keyword argument` problem when upper_right_corner is passed. Would you rather rather replace **kwargs there with e.g. upper_right_corner='native'? I would like to avoid changing the calling line in https://github.com/pytroll/satpy/blob/master/satpy/readers/yaml_reader.py#L667 since I don't know if there are other implications to it.

Reply:
In the same way that the segmented reader just add `pad_data`, you can just add `upper_right_corner`. In principle the api doesn't handle more arguments. Also you don't need `pad_data` here as it's captured by the daugther class already.

```suggestion
    def _load_dataset(dsid, ds_info, file_handlers, dim='y', upper_right_corner='native'):
```

Reply:
@ameraner made me realise my comment is wrong (I was thinking this was in `GEOSegmentYAMLReader`). So `pad_data` should be there

Reply:
After discussion with Martin I made `upper_right_corner` explicit in `_load_dataset_with_area`, which avoids the need of the extra **kwargs. `pad_data` is still in the **kwargs to keep its default definition at the functions right where it's used.

Reply:
:+1: